### PR TITLE
Get large avatar images as scale and crop locally

### LINKED ISSUE #1069: Avatar is blurry in "Room Setting" and "{Global,Room} User Profile" under HiDPI
### Describe the bug

Avatar is blurry in "Room Setting" and "{Global,Room} User Profile" under HiDPI

### To Reproduce

0. Have a scale factor of 2.
1. Open any Room Settings or User Profile.
2. Observe the avatar.

### What happened?

Avatar is blurry.

### Expected behavior

Avatar should be sharp.

### Screenshots

_No response_

### Version

0.9.3-cbb804cb

### Operating system

Linux

### Installation method

Some repository (AUR, homebrew, distribution repository, PPA, etc)

### Qt version

5.15.3

### C++ compiler

gcc-11.2

### Desktop Environment

Repro'd on all of GNOME Xorg and GNOME Wayland and sway

### Did you use profiles?

- [ ] Profiles used?

### Relevant log output

_No response_

### Backtrace

_No response_

### PR DESCRIPTION
**Pull request by brausepulver**
Resolves #1069

The Matrix spec requires servers to provide thumbnails at (96x96, crop) and (320x240, scale) among others. [1] The avatars in Nheko's global/room profile and room settings are sized 130x130 on normal scaling and 260x260 on 2x scaling like on a HiDPI device. In both cases the avatar is requested as cropped and that way displayed at 96x96, making it look blurry.

This can be solved by requesting scaled avatars rather than cropped where appropriate, and cropping to the requested size afterwards.

HiDPI can be simulated in Qt by setting QT_SCALE_FACTOR=2.

[1] https://spec.matrix.org/v1.3/client-server-api/#thumbnails

### CODE REVIEW

File: src/MxcImageProvider.cpp [hunk 1]
```diff
@@ -97,6 +97,14 @@ clipRadius(QImage img, double radius)
     return out;
 }
 
+QImage
+MxcImageProvider::cropImage(const QImage &image)
```
Comment (deepbluev7) [inline src/MxcImageProvider.cpp:101]: How about instead you do:

```
auto temp = image =
                        image.scaled(requestedSize, Qt::KeepAspectRatioByExpanding, Qt::SmoothTransformation);
scaled = temp.copy((temp.width() - requestedSize.width())/2, (temp.height() - requestedSize.height())/2, requestedSize.width(), requestedSize.height());
```

Comment (brausepulver) [reply src/MxcImageProvider.cpp]: Yep, that seems to work!

**Comment by deepbluev7** [comment]
Seems to work well for me, thank you for the contribution! Although I use dynamic thumbnailing, so I am relying on your statement that it works for you as well :3

**Comment by brausepulver** [comment]
It does yeah :) How do you get dynamic thumbnailing?

**Comment by deepbluev7** [comment]
It is a config option in synapse.

**Comment by brausepulver** [comment]
Neat thanks. The matrix.org homeserver doesn't seem to support this at least.

**Pull request merged**